examples/credential-registry-storage-contract/src/lib.rs

```diff
@@ -0,0 +1,404 @@
+#[derive(Serialize, SchemaType)]
+pub struct StoreParam {
+    /// Signature.
+    signature:  SignatureEd25519,
+    /// PublicKeyEd25519 that created the above signature.
+    public_key: PublicKeyEd25519,
+    /// Message that was signed.
+    message:    SignedMessage,
+}
+
+#[derive(Serialize)]
+pub struct PartialStoreParam {
+    /// Signature.
+    signature:  SignatureEd25519,
+    /// PublicKeyEd25519 that created the above signature.
+    public_key: PublicKeyEd25519,
+}
+
+/// Store credential data.
+/// It rejects if:
+/// - It fails to parse the parameter.
+/// - The signature was intended for a different contract.
+/// - The signature was intended for a different `entry_point`.
+/// - The signature is expired.
+/// - The signature can not be validated.
+/// - A credential is already registered for the given `public_key`.
+/// - Fails to log event.
+#[receive(
+    contract = "credential-registry-storage",
+    name = "store",
+    parameter = "StoreParam",
+    error = "CustomContractError",
+    crypto_primitives,
+    enable_logger,
+    mutable
+)]
+fn store<S: HasStateApi>(
+    ctx: &impl HasReceiveContext,
+    host: &mut impl HasHost<State<S>, StateApiType = S>,
+    logger: &mut impl HasLogger,
+    crypto_primitives: &impl HasCryptoPrimitives,
+) -> ContractResult<()> {
+    // Parse the parameter.
+    let mut cursor = ctx.parameter_cursor();
+    // The input parameter is `StoreParam` but we only read the initial part of it
+    // with `PartialStoreParam`. I.e. we read the `signature` and the
+    // `public_key`, but not the `message` here.
+    let param: PartialStoreParam = cursor.get()?;
+
+    // We read in the `message` now. `(cursor.size() - cursor.cursor_position()` is
+    // the length of the message in bytes.
+    let mut message_bytes = vec![0; (cursor.size() - cursor.cursor_position()) as usize];
+
+    cursor.read_exact(&mut message_bytes)?;
+
+    // The plan is to develop the signature schema specific to verifiable
+    // credentials in the future but for the time being, we use the signature schema
+    // that is already implemented in the Concordium browser wallet. The message
+    // signed in the Concordium browser wallet is prepended with the `account`
+    // address and 8 zero bytes. Accounts in the Concordium browser wallet can
+    // either sign a regular transaction (in that case the prepend is `account`
+    // address and the nonce of the account which is by design >= 1) or sign a
+    // message (in that case the prepend is `account` address and 8 zero bytes).
+    // Hence, the 8 zero bytes ensure that the user does not accidentally sign a
+    // transaction. The account nonce is of type u64 (8 bytes).
+    let mut msg_prepend = vec![0; 32 + 8];
+    // Prepend the `account` address of the signer.
+    msg_prepend[0..32].copy_from_slice(&param.public_key.0);
+    // Prepend 8 zero bytes.
+    msg_prepend[32..40].copy_from_slice(&[0u8; 8]);
```

Comment:
This is not necessary, but we should instead prepend `WEB3ID:STORE`. That's the chosen prefix.